protocol/messenger_mailserver.go

```diff
@@ -85,73 +82,49 @@ func (m *Messenger) scheduleSyncChat(chat *Chat) (bool, error) {
 				m.config.messengerSignalsHandler.MessengerResponse(response)
 			}
 			return response, nil
-		})
+		}, history.WithPeerID(peerID))
 		if err != nil {
 			m.logger.Error("failed to perform mailserver request", zap.Error(err))
 		}
 	}()
 	return true, nil
 }
 
-func (m *Messenger) connectToNewMailserverAndWait() error {
-	// Handle pinned mailservers
-	m.logger.Info("disconnecting mailserver")
-	pinnedMailserver, err := m.getPinnedMailserver()
-	if err != nil {
-		m.logger.Error("could not obtain the pinned mailserver", zap.Error(err))
-		return err
-	}
-	// If pinned mailserver is not nil, no need to disconnect and wait for it to be available
-	if pinnedMailserver == nil {
-		m.disconnectActiveMailserver(graylistBackoff)
-	}
+func (m *Messenger) performStorenodeTask(task func() (*MessengerResponse, error), opts ...history.StorenodeTaskOption) (*MessengerResponse, error) {
```

Comment:
This wrapper looks quite difficult. We start a goroutine that has a callback that calls a task, whilst receiving result/error through channels 😵

Any chances to simplify this somehow?

Reply:
Should be simpler now!